### ISSUE TITLE
Issue setting user:pass authenticated session proxy (still)

### ISSUE THREAD

**Issue by theyezzyman** [electron/electron#14589]
**Expected Behavior**
Use the session.setProxy function of a window to set a user:pass authenticated proxy. Load URL in that window. Expect that app.login event is triggered and request is made.
**Actual behavior**
Set session proxy to a user:pass authenticated proxy, loadURL requests fail with status "failed" in dev tools. App.login event never called.
**To Reproduce**
Create a window, use the session setProxy function to set a user:pass authenticated proxy. Then try to load ANY url in that window.

```
win.webContents.session.setProxy({
          proxyRules: `http://user:pass@ip:port`
        }, () => {});
 }
//Requests fail status "failed"
win.loadURL('https://whatismyipaddress.com');
//This is NOT reached
app.on('login', function (event, request, authInfo, callback) {
  if (authInfo.isProxy) {
    log.info(`HI ${JSON.stringify(authInfo)}`)
    // callback('username', 'password');
  }
})
```
Working code with IP authenticated proxies
```
win.webContents.session.setProxy({
          proxyRules: `http://ip:port`
        }, () => {});
 }
//Requests succeed
win.loadURL('https://whatismyipaddress.com');
//This is reached
app.on('login', function (event, request, authInfo, callback) {
  if (authInfo.isProxy) {
    log.info(`HI ${JSON.stringify(authInfo)}`)
    // callback('username', 'password');
  }
})
```

This code is working for me with IP authenticated proxies. App.login is triggered and the request completes as expected. Just user:pass proxies that are the issue.

When opening chrome dev tools I can see the requests being attempted and returning status "failed".

**Additional Information**

I opened this issue and it was closed, marked as a duplicate of https://github.com/electron/electron/issues/12443

This is a different case and not a duplicate. Please do not close this! Thank you.

*edited*
- Electron version :2.0.1
- Platform: MacOS High Sierra 10.13.3

**Comment by codebytere** [comment]
@theyezzyman there is no information here about platform, version, etc. Please add that before we can begin to address this issue.

**Comment by theyezzyman** [comment]
@codebytere updated, thanks

**Comment by theyezzyman** [comment]
```
const { app, BrowserWindow } = require('electron');

let mainWin;

let proxies = { //append all proxies to this object to allow for user/pass lookup
    '181.214.51.93:21605': {
        username: 'chi',
        password: 'chi09',
    }
};

app.on('ready', () => {
    mainWin = new BrowserWindow({show:true});

    mainWin.webContents.session.setProxy('181.214.51.93:21605', () => { //the proxy host value here would be fed in from the existing proxy lib file
        mainWin.loadURL('https://www.whatismyip.com/'); //PoC to show it actually uses the proxy
    });
});

app.on('login', (event, webContents, request, authInfo, callback) => {
    let fullProxy = `${authInfo.host}:${authInfo.port}`; //concat proxy for lookup

    callback(proxies[fullProxy].username, proxies[fullProxy].password); //supply credentials to server
});
```

Solution found if anyone runs into the same issue I did

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v4.2.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.

**Comment by timothymiller** [comment]
I am experiencing this issue on Electron v6.0.0

The expected call to app.on('login' is never called.

**Comment by MatthieuLeMee** [comment]
Any update on this one ?
Missing proxy auth is really blocking for enterprise use...

**Comment by nornagon** [comment]
Duplicate of https://github.com/electron/electron/issues/20752

**Comment by MatthieuLeMee** [comment]
This is not a duplicate of #20752 as this one is about authentication for network proxy and the other issue is about basic auth on websites.

**Comment by phpproff** [comment]
same issue electron 7
missing proxy auth

**Comment by dafivius** [comment]
Same issue on Electron 8, no auth. Proxy without authorization works well.

**Comment by robsonkades** [comment]
Did anyone succeed?

**Comment by mattlm0831** [comment]
I have been having this issue for almost a year at this point, with no working solution.

**Comment by robsonkades** [comment]
Here it worked with a simple proxy, my problem was with the electron update.

@mattlm0831 
```
const proxies = {
  '127.0.0.1:8888': {
    username: '1',
    password: '1',
  },
};

autoUpdater.on('login', async (authInfo, callback) => {
  const fullProxy = `${authInfo.host}:${authInfo.port}`;
  callback(proxies[fullProxy].username, proxies[fullProxy].password);
});

```

**Comment by restocked** [comment]
Did anyone found a solution to this issue?

**Comment by kunokdev** [comment]
> Here it worked with a simple proxy, my problem was with the electron update.
> 
> @mattlm0831
> 
> ```
> const proxies = {
>   '127.0.0.1:8888': {
>     username: '1',
>     password: '1',
>   },
> };
> 
> autoUpdater.on('login', async (authInfo, callback) => {
>   const fullProxy = `${authInfo.host}:${authInfo.port}`;
>   callback(proxies[fullProxy].username, proxies[fullProxy].password);
> });
> ```

Why are you using `autoUpdater`? If I change `app.on('login')` `autoUpdater.on('login')` I get error that 'login' event doesn't exist on type 'update-is-not-available'

**Comment by hellpain1412** [comment]
Did anyone found a solution to this issue?

**Comment by linzixin-muzi** [comment]
How does socks5 support
